newScript.js

```diff
@@ -0,0 +1,295 @@
+    if (i % 2 === 0) {
+      listToDo.splice(i, 1);
+    }
+  }
+  updatePage();
+})
+
+removeDup.addEventListener("click", () => {
+  const tempSet = new Set();
+  const tempArray = [];
+  for (let i = 0; i < listToDo.length; i++) {
+    if (!tempSet.has(listToDo[i].taskName)) {
+      tempArray.push(listToDo[i]);
+    }
+    tempSet.add(listToDo[i].taskName);
+  }
+  listToDo = tempArray;
+  updatePage();
+});
+
+removeRandom.addEventListener("click", () => {
+  const random = Math.round(Math.random() * listToDo.length - 1);
+  listToDo.splice(random, 1);
+  updatePage();
+});
+
+removeLast.addEventListener("click", () => {
+  listToDo.pop();
+  updatePage();
+});
+
+removeFirst.addEventListener("click", () => {
+  listToDo.splice(0, 1);
+  updatePage();
+});
+
+addRandom.addEventListener("click", () => {
+  const randomNum = Math.round(Math.random() * listToDo.length - 1);
+  const newItem = {
+    taskName: input.value,
+    date: dateInput.value,
+    tag: tagInput.value
+  }
+  listToDo.splice(randomNum, 0, newItem);
+  updatePage();
+});
+
+addFirstBtn.addEventListener("click", () => {
+  const newItem = {
+    taskName: input.value,
+    date: dateInput.value,
+    tag: tagInput.value
+  }
+  listToDo.unshift(newItem);
+  updatePage();
+});
+
+randomize.addEventListener("click", () => {
+  listToDo.sort(() => Math.random() - 0.5);
+  updatePage();
+});
+
+alphaBtn.addEventListener("click", () => {
+  listToDo.sort((a, b) => {
+    const taskA = a.taskName.toUpperCase();
+    const taskB = b.taskName.toUpperCase();
+    if (taskA > taskB) {
+      return 1;
+    }
+    if (taskA < taskB) {
+      return -1;
+    }
+    return 0;
+  });
+  updatePage();
+});
+
+alphaRvrsBtn.addEventListener("click", () => {
+  listToDo.sort((a, b) => {
+    const taskA = a.taskName.toUpperCase();
+    const taskB = b.taskName.toUpperCase();
+    if (taskA > taskB) {
+      return -1;
+    }
+    if (taskA < taskB) {
+      return 1;
+    }
+    return 0;
+  });
+
+  updatePage();
+});
+
+window.addEventListener("load", () => {
+  for (let i = 0; i < listToDo.length; i++) {
+    createOnPg(listToDo[i].taskName, listToDo[i].date, listToDo[i].tag);
+  }
+});
+
+function refreshLocalStorage(updatedList) {
+  localStorage.setItem("list", JSON.stringify(updatedList));
+};
+
+seeMoreBtn.addEventListener("click", (e) => {
+  e.preventDefault();
+  if (dateInput.getAttribute("id") === "date-appear") {
+    dateInput.setAttribute("id", "date-disappear");
+    setTimeout(() => { tagInput.setAttribute("id", "tag-disappear") }, 100);
+    setTimeout(() => {
+      tagInput.setAttribute("id", "tag");
+      dateInput.setAttribute("id", "date");
+    }, 400);
+  }
+
+  if (seeMoreDiv.style.display === "block") {
+    seeMoreDiv.style.display = "none"
+  } else {
+    seeMoreDiv.style.display = "block"
+  }
+});
+
+subBtn.addEventListener("click", () => {
+  if (!input.value) {
+    window.alert("You're adding an empty task!");
+    return
```

Comment:
Pormenores de consistência:

```suggestion
    return;
```